src/query/exceptions.hpp

```diff
@@ -211,6 +211,21 @@ class IsolationLevelModificationInMulticommandTxException : public QueryExceptio
       : QueryException("Isolation level cannot be modified in multicommand transactions.") {}
 };
 
+class IsolationLevelModificationInAnalyticsException : public QueryException {
+ public:
+  IsolationLevelModificationInAnalyticsException()
+      : QueryException(
+            "Isolation level cannot be modified when storage mode is set to IN_MEMORY_ANALYTICAL."
+            "IN_MEMORY_ANALYTICAL mode doesn't provide any isolation guarantees, "
+            "you can think about it as an equivalent to READ_UNCOMMITED.") {}
```

Comment:
Sorry for being picky but I think this last sentence should be a bit changed because I find think it is a bit weird to have "you can think about" in the exception message. I think "IN_MEMORY_ANALYTICAL mode doesn't provide any isolation guarantees. " is enough. Or: ... provide any iso guarantees, it is an equivalent to READ_UNCOMMITTED, to avoid explicit message to user.